File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = 0;
        for(int i=0;i<s.size();i++){
            string sub = "";
            int hash[256] = {0};
            for(int j=i;j<s.size();j++){
                if(hash[s[j]]) break;
                sub += s[j];
                hash[s[j]] = 1;
                n = max(n,j-i+1);
            }
        }
        return n;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> hmp = new HashSet<>();
        int l = 0;
        int r = 0;
        int maxlen = Integer.MIN_VALUE;
        while(r<s.length()) {
            while(hmp.contains(s.charAt(r))!=false) {
                hmp.remove(s.charAt(l));
                l++;
            }
            hmp.add(s.charAt(r));
            maxlen = max(maxlen,r-l+1);
            r++;
        }
        return (maxlen==Integer.MIN_VALUE) ? 0:maxlen;
    }
    public int max(int a, int b) {
        return (a>b) ? a:b;
    }
}